eis_toolkit/prediction/mlp.py

```diff
@@ -0,0 +1,368 @@
+
+import numpy as np
+from beartype import beartype
+from keras.optimizers import Adam
+from tensorflow import keras
+
+from eis_toolkit import exceptions
+
+
+def _keras_optimizer(optimizer: str, **kwargs):
+    if optimizer == "adam":
+        return Adam(**kwargs)
+    else:
+        raise exceptions.InvalidParameterValueException(f"Unidentified optimizer: {optimizer}")
+
+
+def _check_MLP_inputs(
+    neurons: Sequence[int],
+    validation_split: Optional[float],
+    learning_rate: float,
+    dropout_rate: Optional[Number],
+    es_patience: int,
+    batch_size: int,
+    epochs: int,
+    output_neurons: int,
+    loss_function: str,
+) -> None:
+    """Check inputs for MLP."""
+    if len(neurons) == 0:
+        raise exceptions.InvalidParameterValueException("Neurons parameter must be a non-empty list.")
+
+    if any(neuron < 1 for neuron in neurons):
+        raise exceptions.InvalidParameterValueException("Each neuron in neurons list must be at least 1.")
+
+    if validation_split and not (0 < validation_split < 1):
+        raise exceptions.InvalidParameterValueException("Validation split must be a value between 0 and 1, exclusive.")
+
+    if learning_rate <= 0:
+        raise exceptions.InvalidParameterValueException("Learning rate must be greater than 0.")
+
+    if dropout_rate and not (0 <= dropout_rate <= 1):
+        raise exceptions.InvalidParameterValueException("Dropout rate must be between 0 and 1, inclusive.")
+
+    if es_patience <= 0:
+        raise exceptions.InvalidParameterValueException("Early stopping patience must be greater than 0.")
+
+    if batch_size <= 0:
+        raise exceptions.InvalidParameterValueException("Batch size must be greater than 0.")
+
+    if epochs <= 0:
+        raise exceptions.InvalidParameterValueException("Number of epochs must be greater than 0.")
+
+    if output_neurons <= 0:
+        raise exceptions.InvalidParameterValueException("Number of output neurons must be greater than 0.")
+
+    if output_neurons > 1 and loss_function == "binary_crossentropy":
+        raise exceptions.InvalidParameterValueException(
+            "Number of output neurons must be 1 when used loss function is binary crossentropy."
+        )
+
+
+@beartype
+def train_MLP_classifier(
+    X: np.ndarray,
+    y: np.ndarray,
+    neurons: Sequence[int] = [16],
+    validation_split: Optional[float] = 0.2,
+    activation: Literal["relu"] = "relu",
+    output_neurons: int = 1,
+    last_activation: Literal["softmax", "sigmoid"] = "softmax",
+    epochs: int = 50,
+    batch_size: int = 32,
+    optimizer: Literal["adam"] = "adam",
+    learning_rate: Number = 0.001,
+    loss_function: Literal["categorical_crossentropy", "binary_crossentropy"] = "categorical_crossentropy",
+    dropout_rate: Optional[Number] = None,
+    early_stopping: bool = True,
+    es_patience: int = 5,
+    metrics: Optional[Sequence[Literal["accuracy", "precision", "recall"]]] = ["accuracy"],
```

Comment:
F1score does not seem to be listed at least here https://keras.io/api/metrics/

Reply:
Yeah, I was puzzled why I couldn't find in the docs since it should be a used metric. Odd that it didn't show in the site I looked, but I added it now!